src/App.tsx

```diff
@@ -14,10 +14,24 @@ import Footer from './components/Footer';
 function App() {
   const [themeType, setThemeType] = useState<'light' | 'dark' | undefined>('light');
   const toggleMode = useCallback(() => {
+    if (themeType === 'light') {
+      localStorage.setItem('themeType', 'dark');
+    } else {
+      localStorage.setItem('themeType', 'light');
+    }
     setThemeType(themeType === 'light' ? 'dark' : 'light');
   }, [themeType, setThemeType]);
 
   const theme: Theme = useMemo(() => {
+    if (!localStorage.getItem('themeType')) {
+      if (themeType === 'light') {
+        localStorage.setItem('themeType', 'dark');
+      } else {
+        localStorage.setItem('themeType', 'light');
+      }
+    } else {
+      setThemeType(localStorage.getItem('themeType') === 'light' ? 'light' : 'dark');
+    }
```

Comment:
This should be in a `useEffect` where if the localStorage theme isn't set we set it to default. Also is there any way we can use eventlisteners to simplify some of this code? we can listen for the 'storage' event with `window.addEventListener('storage', ...)` and we check if the localstorage had any changes to do with theme in the 2nd arg